includes/discovery/functions.inc.php

```diff
@@ -145,6 +152,8 @@ function discover_device($device, $options = null)
             echo "#### Unload disco module $module ####\n\n";
         } elseif (isset($attribs['discover_' . $module]) && $attribs['discover_' . $module] == '0') {
             echo "Module [ $module ] disabled on host.\n\n";
+        } elseif (isset($os_module_status) && $os_module_status == '0') {
```

Comment:
And here.  You could probably find a way to only have one `isset($config['os'][$device['os']]['discovery_modules'][$module])`